From a11e3636c8e4e7745d2097ce1675c0e1a9bff5d0 Mon Sep 17 00:00:00 2001
From: lejianwen <84855512@qq.com>
Date: Tue, 21 Jan 2025 19:22:26 +0800
Subject: [PATCH 17/32] Up Version

---
 Cargo.toml   | 2 +-
 ui/setup.nsi | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index d38f86b..2c4209f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "hbbs"
-version = "1.1.13"
+version = "1.1.14"
 authors = ["rustdesk <info@rustdesk.com>"]
 edition = "2021"
 build = "build.rs"
diff --git a/ui/setup.nsi b/ui/setup.nsi
index b48e1ab..fe58a1c 100644
--- a/ui/setup.nsi
+++ b/ui/setup.nsi
@@ -15,7 +15,7 @@
 !define PRODUCT_NAME "rustdesk_server"
 !define PRODUCT_DESCRIPTION "Installer for ${PRODUCT_NAME}"
 !define COPYRIGHT "Copyright © 2021"
-!define VERSION "1.1.13"
+!define VERSION "1.1.14"
 
 VIProductVersion "${VERSION}.0"
 VIAddVersionKey "ProductName" "${PRODUCT_NAME}"
-- 
2.54.0 (Apple Git-157)

